lib/nnue_training_data_formats.h

```diff
@@ -7577,6 +7578,11 @@ namespace binpack
 
                     if (!m_localBuffer.empty())
                     {
+                        // now shuffle the local buffer
+			std::random_device rd;
+			static thread_local std::mt19937 g(rd());
```

Comment:
do static thread_local std::mt19937 g(std::random_device()); so that the random device it's created once

Reply:
doesn't seem allowed ?:
`/home/vondele/chess/vondele/nnue-pytorch/lib/nnue_training_data_formats.h: In lambda function:
/home/vondele/chess/vondele/nnue-pytorch/lib/nnue_training_data_formats.h:7582:11: error: storage class ‘thread_local’ invalid for function ‘g’
 7582 |    static thread_local std::mt19937 g(std::random_device());
      |           ^~~~~~~~~~~~
/home/vondele/chess/vondele/nnue-pytorch/lib/nnue_training_data_formats.h:7582:4: error: cannot declare static function inside another function
 7582 |    static thread_local std::mt19937 g(std::random_device());
      |    ^~~~~~
`

Reply:
right, `static thread_local std::mt19937 g(std::random_device{}());`

Reply:
done